src/commands/CommandClear.java

```diff
@@ -22,31 +24,70 @@
 	@Override
 	public void action(){
 		
-		new CommandConfirmed(this){
+		if(!hasParameter("u")){
 			
-			@Override
-			public String getConfMessage(){
-				return lang("ConfirmMessage");
-			}
-			
-			@Override
-			public void confirmed(){
+			new CommandConfirmed(this){
 				
-				try{
-					
-					deleteAllMessages();
+				@Override
+				public String getConfMessage(){
+					return lang("ConfirmMessage");
+				}
+				
+				@Override
+				public void confirmed(){
 					
-					if(isAlive())
-						sendMessage("Cleared everything!");
+					try{
+						
+						deleteAllMessages();
+						
+						if(isAlive())
+							sendMessage("Cleared everything!");
+						
+					}
+					catch(PermissionException e){
+						sendMessage(lang("NoPermission"));
+					}
 					
 				}
-				catch(PermissionException e){
-					sendMessage(lang("NoPermission"));
-				}
 				
+			};
+		}
+		else{
+			
+			try{
+				final Member usr = getParameterAsMention("u");
```

Comment:
I'd do a if-else blocks here to determine which parameter exists (in order : `-u`, `--self`, `--bot`) and add the user to the member object afterward.